test/unit-test/MQTTv5/core_mqttv5_serializer_utest.c

```diff
@@ -0,0 +1,3299 @@
+    connectInfo.clientIdentifierLength = UINT16_MAX;
+    connectInfo.pPassword = "";
+    connectInfo.passwordLength = UINT16_MAX;
+    connectInfo.pUserName = "";
+    connectInfo.userNameLength = UINT16_MAX;
+    willInfo.pTopicName = TEST_TOPIC_NAME;
+    willInfo.topicNameLength = UINT16_MAX;
+    /* A valid will message payload is less than the maximum 16 bit integer. */
+    willInfo.payloadLength = UINT16_MAX + 2;
+    status = MQTTV5_GetConnectPacketSize( &connectInfo, &willInfo, &properties, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, status );
+
+    /* Verify good case */
+    memset( &connectInfo, 0x0, sizeof( connectInfo ) );
+    connectInfo.cleanSession = true;
+    connectInfo.pClientIdentifier = "TEST";
+    connectInfo.clientIdentifierLength = 4;
+    status = MQTTV5_GetConnectPacketSize( &connectInfo, NULL, &properties, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    TEST_ASSERT_EQUAL_INT( 0, properties.propertyLength );
+    /* Make sure remaining size returned is 17. */
+    TEST_ASSERT_EQUAL_INT( 17, remainingLength );
+    /* Make sure packet size is 19. */
+    TEST_ASSERT_EQUAL_INT( 19, packetSize );
+
+    /* With will. These parameters will cause the packet to be
+     * 4 + 2 + 8 + 2 = 16 bytes larger. */
+    memset( &willInfo, 0x0, sizeof( willInfo ) );
+    willInfo.pTopicName = "test";
+    willInfo.topicNameLength = 4;
+    willInfo.pPayload = "testload";
+    willInfo.payloadLength = 8;
+    status = MQTTV5_GetConnectPacketSize( &connectInfo, &willInfo, &properties, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    /* Make sure remaining size returned is 32 = 16 + 16 + 2. */
+    TEST_ASSERT_EQUAL_INT( 34, remainingLength );
+    /* Make sure packet size is 34 = 18 + 16 + 2. */
+    TEST_ASSERT_EQUAL_INT( 36, packetSize );
+
+    /* With username and password. This will add 4 + 2 + 4 + 2 = 12 bytes. */
+    connectInfo.cleanSession = true;
+    connectInfo.pUserName = "USER";
+    connectInfo.userNameLength = 4;
+    connectInfo.pPassword = "PASS";
+    connectInfo.passwordLength = 4;
+    status = MQTTV5_GetConnectPacketSize( &connectInfo, NULL, &properties, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL( MQTTSuccess, status );
+    /* Make sure remaining size returned is 28 = 16 + 12. */
+    TEST_ASSERT_EQUAL_INT( 29, remainingLength );
+    /* Make sure packet size is 30 = 18 + 12. */
+    TEST_ASSERT_EQUAL_INT( 31, packetSize );
+
+    status = MQTTV5_GetConnectPacketSize( &connectInfo, NULL, NULL, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+
+    /*Validating the function Get Connect properties*/
+
+    /* Call MQTT_GetConnectPropertiesSize() with various combinations of
+     * incorrect paramters */
+
+    /*
+     * Max Packet Size cannot be null
+     */
+    properties.receiveMax = 24;
```

Comment:
what about tests for other properties like session expiry? need tests for max bounds as well I guess?  I think MQTT the GetConnectPropertiesSize needs further checks for testing MAX values of sessionexpiry, receive max and maxpacket size

Reply:
Max possible value of session expiry is uint32 max, receive max is uint16 max and  max packet size is 256 MB but even if  we set it beyond that it doesn't matter because it is theoretically impossible for max packet size to be more than this because of the limitation of the encoded remaining length.